tests/test-send-lists.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Class Newsletters Test Send_Lists
+ *
+ * @package Newspack_Newsletters
+ */
+
+use Newspack\Newsletters\Send_List;
+use Newspack\Newsletters\Send_Lists;
+
+/**
+ * Tests the Send_List class
+ */
+class Send_Lists_Test extends WP_UnitTestCase {
+
+	use Send_Lists_Setup;
+
+	/**
+	 * Test matching by item ID.
+	 */
+	public function test_match_by_id() {
+		$this->assertTrue( Send_Lists::matches_id( '123', '123' ) ); // Single ID matches.
+		$this->assertTrue( Send_Lists::matches_id( [ '123', '456' ], '123' ) ); // Array of IDs matches.
+		$this->assertFalse( Send_Lists::matches_id( '456', '123' ) ); // Single of ID doesn't match.
+		$this->assertFalse( Send_Lists::matches_id( [ '456', '789' ], '123' ) ); // Array of IDs doesn't match.
+	}
+
+	/**
+	 * Test matching by search term(s).
+	 */
+	public function test_match_by_search() {
```

Comment:
Nice!

When testing methods like this it is always desirable to test some reallly unhappy tests.

For example, what's the expected return if I pass NULL as one of the parameters? or an Object of any kind? or an integer, etc.

Reply:
Great idea! f9bd959 adds some of these tests (as well as some missing handling for those weird cases!).

Reply:
That's the beauty of writing tests :)